### PR TITLE
Add absolute file path detection

### PR DESCRIPTION
**Pull request by Elscrux**
This part will help detecting string like those: `C:\Skyrim\Data\Meshes\Clutter\MyMesh.nif` and turn them into `Meshes\Clutter\MyMesh.nif`.

### CODE REVIEW

File: Mutagen.Bethesda.Core/Plugins/Assets/AssetLink.cs [hunk 1]
```diff
@@ -39,17 +41,31 @@ public AssetLinkGetter(string rawPath)
 
     public string DataRelativePath
     {
-        get
-        {
-            var trimmedPath = RawPath
+        get {
```
Comment (Noggog) [inline Mutagen.Bethesda.Core/Plugins/Assets/AssetLink.cs]: If you can run this code through the code formatter that'd be good.  For the repos here, id like to keep it to the C# styling, so future devs don't get too crazy of diffs when they do any auto-formatting

File: Mutagen.Bethesda.Core/Plugins/Assets/AssetLink.cs [hunk 2]
```diff
@@ -39,17 +41,31 @@ public AssetLinkGetter(string rawPath)
...
+            var path = RawPath;
+            
+            // Reduce all absolute paths to the path under data directory
+            if (path.Contains(Path.VolumeSeparatorChar)) {
```
Comment (Noggog) [inline Mutagen.Bethesda.Core/Plugins/Assets/AssetLink.cs]: Maybe refactor out to a private function or two, for clarity?  Getting pretty complex for a property definition, imo.

File: Mutagen.Bethesda.Core/Plugins/Assets/IAssetLink.cs [hunk 3]
```diff
@@ -14,11 +14,16 @@ public interface IAssetLinkGetter
...
     /// <summary>
-    /// Raw path relative to the game's data directory, as it is commonly known in the Skyrim modding community.
+    /// Raw path relative to the game's data directory, as it is commonly known in the Bethesda modding community.
+    /// <example>
+    /// <code>
     /// Example:
-    ///     \Data\Meshes\Clutter\MyMesh.nif => Meshes\Clutter\MyMesh.nif
-    ///     Data\Meshes\Clutter\MyMesh.nif  => Meshes\Clutter\MyMesh.nif
-    ///     Clutter\MyMesh.nif              => Meshes\Clutter\MyMesh.nif
+    ///     C:\Skyrim\Data\Meshes\Clutter\MyMesh.nif => Meshes\Clutter\MyMesh.nif
```
Comment (Noggog) [inline Mutagen.Bethesda.Core/Plugins/Assets/IAssetLink.cs:21]: I think this DataRelativePath is complex enough and has arbitrary and important enough logic to warrant unit tests.   I thought we'd have had some in the AssetLinkTests class, but not seeing any specifically related to testing DataRelativePath too much.

I think if we can get at least 4 tests, one for each standardization style listed here, that'd be enough?

**Review by Noggog — CHANGES_REQUESTED** [review]

File: Mutagen.Bethesda.Core/Plugins/Assets/AssetLink.cs (hunk 1 above)
Comment (Elscrux) [reply Mutagen.Bethesda.Core/Plugins/Assets/AssetLink.cs]: oops, yeah I missed that

**Pull request merged**